### PR TITLE
Regrade papers

### PR DESCRIPTION
**Pull request by prathamesh920**
A question or whole paper can be regraded.
We can regrade as per user answer paper.

**In Next PR**
I will be adding more testcases and will clean up the code further.
Also, will make the UI much better.

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -14,10 +14,17 @@
...
+from os.path import join, abspath, dirname, exists
 import shutil
 import zipfile
 import tempfile
 from file_utils import extract_files
+from yaksh.xmlrpc_clients import code_server
+
+
+# The directory where user data can be saved.
+OUTPUT_DIR = abspath(join(dirname(__file__), 'output'))
```
Comment (prabhuramachandran) [inline yaksh/models.py]: This is not good, what if yaksh is installed in a non-writeable location.

File: yaksh/models.py [hunk 2]
```diff
@@ -994,6 +1013,71 @@ def get_previous_answers(self, question):
         if question.type == 'code':
             return self.answers.filter(question=question).order_by('-id')
 
+    def validate_answer(self, user_answer, question, json_data=None):
```
Comment (ankitjavalkar) [inline yaksh/models.py]: I guess, the "question" argument is not required.

**Comment by ankitjavalkar** [comment]
Rest of the code looks good

File: yaksh/models.py (hunk 1 above)
Comment (prathamesh920) [inline yaksh/models.py]: resolved

**Pull request merged**